lib/simplifyExpression/basicsSearch/basicRules.js

```diff
@@ -0,0 +1,224 @@
+const {build, query} = require('math-nodes');
+const {rewriteNode} = require('math-rules');
+
+const ChangeTypes = require('../../ChangeTypes');
+const checks = require('../../checks');
+const defineRuleString = require('../../util/defineRuleString');
+const Node = require('../../node');
+
+// e.g. -(-3) -> 3
+const NEGATION = defineRuleString('--#a', '#a');
+
+// Simplifies two unary minuses in a row by removing both of them.
+// e.g. -(- 4) --> 4
+function negation(node) {
+  const newNode = rewriteNode(NEGATION, node);
+  if (newNode) {
+    return Node.Status.nodeChanged(
+      ChangeTypes.NEGATION, node, newNode);
+  }
+  else {
+    return Node.Status.noChange(node);
+  }
+}
+
+//const REARRANGE_COEFF = defineRuleString('#b * #a', '#a #b', {a: query.isNumber, b: isPolynomialTerm})
+
+// e.g. 2/-1 -> -2
+const DIVISION_BY_NEGATIVE_ONE = defineRuleString('#a / -1', '-#a');
+
+// e.g. 2/1 -> 2
+const DIVISION_BY_ONE = defineRuleString('#a / 1', '#a');
+
+
+// If `node` is a division operation of something by 1 or -1, we can remove the
+// denominator. Returns a Node.Status object.
+function removeDivisionByOne(node) {
+  let newNode = null;
+  newNode = rewriteNode(DIVISION_BY_ONE, node);
+  if (newNode) {
+    return Node.Status.nodeChanged(
+      ChangeTypes.DIVISION_BY_ONE, node, newNode);
+  }
+
+  const numerator = node.args[0];
```

Comment:
you can't assume a node will have `args`, and if `args` is null, `args[0] `will throw an error